deepmd/utils/argcheck.py

```diff
@@ -2311,6 +2311,25 @@ def gen_args(**kwargs) -> List[Argument]:
     ]
 
 
+def backend_compat(data):
+    data = data.copy()
+    # stripped_type_embedding in old DescrptSeAtten
+    if (
+        "descriptor" in data["model"]
+        and data["model"]["descriptor"].get("type", "se_e2_a") == "se_atten"
+        and data["model"]["descriptor"].pop("stripped_type_embedding", False)
+    ):
```

Comment:
It does not support the hybrid descriptor.

Reply:
Perhaps support and add a customized normalize hook to dargs

Reply:
I tried to add `extra_check` in Argument (outside in the descriptor args level), it works but there are still problems:
1. `hybrid` descriptor still need to be individually considered.
2. This keeps `stripped_type_embedding` in `descrpt_se_atten_args`. In other places (such as tests in `source/tests/consistent/descriptor`) where using list of `descrpt_args_plugin`, the `extra_check` can not be reused.

Reply:
> `hybrid` descriptor still need to be individually considered.

It looks to me `hybrid` is defined here, where `extra_check` may be applied.
https://github.com/deepmodeling/deepmd-kit/blob/f3056ee5003b2d33a1394857ee380c0364f405f4/deepmd/utils/argcheck.py#L382-L395

I tried to move `extra_check` to the variant choice argument, i.e., Argument for `se_atten`, but it seems to me the dargs package never executes it. @y1xiaoc any idea about it?

> This keeps `stripped_type_embedding` in `descrpt_se_atten_args`. In other places (such as tests in `source/tests/consistent/descriptor`) where using list of `descrpt_args_plugin`, the `extra_check` can not be reused.

We may deprecate the usage of the `list[Argument]`.

Reply:
It's been so long I almost forgot everything lol. But you're right, the choice argument is just dummy argument as a collection of its sub arguments, no real check is performed at that level. In the checking process it will be flattened and checked at the parent level (at the argument that defines all the sub variants). So maybe you want to add an extra check there.